src/main/java/no/uib/inf112/core/map/MapHandler.java

```diff
@@ -103,16 +103,14 @@
      * @return A read-only set of known robots in the order they were added
      */
     @NotNull
-    Set<Entity> getEntities();
+    Map<Entity, Vector2Int> getEntities();
```

Comment:
I have no issue changing it, just think the orignal author for this solution(@kh498 ) should accept the change 👅

Reply:
We are using the map to keep track of the old position for a entity so that the tile can be removed and then added. If there is a better solution i propose making an issue for it. see [tiledMapHandler](https://github.com/inf112-v19/Blodige-roboter/blob/master/src/main/java/no/uib/inf112/core/map/TiledMapHandler.java#L91-L113)

Reply:
I see that since i made the update class a different class than entities, it needs to be accessed as a map in tiledMapHandler and therefore i made it a map. Alternative is to make the entire variable protected so that tiledMapHandler can acess it. Or create another get wich gets the map and is not a part of the MapHandler interface

Reply:
The whole point of this method is to get all entities on the board. The vectors are cached positions of the entity and is not intended to be seen or changed outside the implementation. Please revert it back to returning a set